src/main/java/org/sonarlint/intellij/SonarApplication.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij;
+
+import com.intellij.ide.plugins.IdeaPluginDescriptor;
+import com.intellij.ide.plugins.PluginManager;
+import com.intellij.lang.Language;
+import com.intellij.lang.LanguageExtensionPoint;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.components.ApplicationComponent;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.extensions.Extensions;
+import com.intellij.openapi.extensions.PluginId;
+import org.jdesktop.swingx.action.ActionManager;
+import org.jetbrains.annotations.NotNull;
+import org.sonarlint.intellij.marker.SonarExternalAnnotator;
+import org.sonarlint.intellij.util.ResourceLoader;
+
+import java.io.IOException;
+
+public class SonarApplication implements ApplicationComponent {
+    private static final Logger LOGGER = Logger.getInstance(SonarApplication.class);
+    private static final String ICON_NAME = "onde-sonar-16.png";
+    private IdeaPluginDescriptor plugin;
+
+    @Override
+    public void initComponent() {
+        plugin = PluginManager.getPlugin(PluginId.getId("org.sonarlint.idea"));
+        for (Language language : Language.getRegisteredLanguages()) {
+            registerExternalAnnotatorFor(language);
+        }
+        //setActionGroupsIcon();
+    }
+
+    private static void setActionGroupsIcon() {
```

Comment:
I left some stuff for future reference (if we want to implement actions on resources, or the auto triggers back), but I'll move it to a branch